free-1.md

```diff
@@ -1,68 +1,98 @@
+calories.
 
 ### Domain
 _Describe the project's domain in five words._
 
+customized planning of diet meals
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+I vision that the user would be led to define their diet as a variable and call
+different functions on the variable to specify different settings like diatary restrictions. Some of the basic fucntions would be something "remove" to remove any ingredients that the user is allergic to and "cut_half" to cut things like carbs in half of what they usually consume.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+Perhaps, the program can be run on terminal and at the beginning, it will list
+all the available functions. First, the user would be asked to name their diet, therefore, defining a variable, and then call different functions on the 
+variable (this can be done one by one, or in a batch mode). And when the user calls a function that indicates the end of setting up, the program will return specific instructions or meal plans for them. 
+
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+Things that are easy to do in this langauge are specifying different settings for the users' meals and, unlike any other apps of this sort, to have different settings for breakfast, lunch, and dinner, or different days of a week. 
+Something that would be difficult is to track their meals. Since there are a lot of apps that allow the user to record their meals, I feel that there's no need for this DSL to do so. This DSL's featuers will mostly focus on helping the user do something about their diet/meals beforehand, rather than afterwards. 
+
 
 ### Related work
 _Are there any other DSLs in this domain? If not, describe how you know there
-aren't and conjecture why not. If so, describe them and provide links. How well 
-do they address the need? Are there any particularly admirable qualities of the
-language? Are there parts of the language you think could be improved?_
+aren't and conjecture why not. If so, describe them and provide links. How well do they address the need? Are there any particularly admirable qualities of thelanguage? Are there parts of the language you think could be improved?_
 
+One of the apps I mentioned I tried out is "myfitnesspal" (https://www.myfitnesspal.com/), which is essnetially a calorie counter. It makes it easy for the user to track what they have been eating and count calories they've been consuming. However, again, it does not let the user plan meals beforehand or evaluate how they are doing on their diet based on anything other than number of calories.
```

Comment:
While I think a DSL could be extremely useful in fine-tuning a diet, the thought of defining a diet as a program frightens me a bit. When a user is programming a server configuration, although errors in code could create headaches, someone's health is not in immediate danger. What happens if someone programs a buggy diet? A diet that is unhealthy purposefully or accidentally? This puts the user at an extreme health risk.

I don't think its an easy thing to decide the suitability of a diet for a purpose and is, perhaps, one of the main reasons why there aren't meal planning apps&mdash;taking on that responsibility is risky and without the insight of a dietician or a physician could put people in immediate danger.